src/evaluate/evaluator/text_classification.py

```diff
@@ -0,0 +1,152 @@
+                - `"simple"` - we evaluate the metric and return the scores.
+                - `"bootstrap"` - on top of computing the metric scores, we calculate the confidence interval for each
+                of the returned metric keys, using `scipy`'s `bootstrap` method
+                https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bootstrap.html.
+            confidence_level (`float`, defaults to `0.95`):
+                The `confidence_level` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            n_resamples (`int`, defaults to `9999`):
+                The `n_resamples` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            random_state (`int`, *optional*, defaults to `None`):
+                The `random_state` value passed to `bootstrap` if `"bootstrap"` strategy is chosen. Useful for
+                debugging.
+            input_column (`str`, defaults to `"text"`):
+                the name of the column containing the text feature in the dataset specified by `data`.
+            label_column (`str`, defaults to `"label"`):
+                the name of the column containing the labels in the dataset specified by `data`.
+            label_mapping (`Dict[str, Number]`, *optional*, defaults to `None`):
+                We want to map class labels defined by the model in the pipeline to values consistent with those
+                defined in the `label_column` of the `data` dataset.
+        Return:
+            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
+            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
+            containing the score, the confidence interval and the standard error calculated for each metric key.
+        Examples:
+        ```python
+        >>> from evaluate import evaluator
+        >>> from datasets import Dataset, load_dataset
+        >>> e = evaluator("text-classification")
+        >>> data =  Dataset.from_dict(load_dataset("imdb")["test"][:2])
+        >>> results = e.compute(
+        >>>     model_or_pipeline="huggingface/prunebert-base-uncased-6-finepruned-w-distil-mnli",
+        >>>     data=data,
+        >>>     metric="accuracy",
+        >>>     input_column="text",
+        >>>     label_column="label",
+        >>>     label_mapping={"LABEL_0": 0.0, "LABEL_1": 1.0},
+        >>>     strategy="bootstrap",
+        >>>     n_resamples=10,
+        >>>     random_state=0
+        >>> )
+        ```"""
+        data = self.prepare_data(data=data, input_column=input_column, label_column=label_column)
+
+        pipe = self.prepare_pipeline(model_or_pipeline=model_or_pipeline, preprocessor=tokenizer)
+
+        metric = self.prepare_metric(metric)
+
+        references = data[label_column]
+        predictions = self._compute_predictions(pipe, data[input_column], label_mapping=label_mapping)
```

Comment:
given current refactoring it might me more readable to do `_compute_predictions` inline (same for `image_classification` - splitting it into a separate function may have been a suboptimal idea from the get go.

Reply:
That's a good point, let me know if the modifs I did are fine.